racket/src/thread/schedule.rkt

```diff
@@ -113,6 +114,14 @@
     (thread-did-work!))
   did?)
 
+;; Check if main thread should collect garbage
+(define (check-garbage-collection)
+  (when (and (= 0 (get-pthread-id)) (chez:collect-garbage-pending?))
```

Comment:
Is the `(= 0 (get-pthread-id))` check needed here? I'd expect the scheduler to run only in the main thread.

Reply:
I wasn't sure about how threads and futures interacted and if futures could run their own threads independent from the main racket thread. If the answer is they cannot, then yes that check is unnecessary.